integrationtests/src/test/java/org/axonframework/integrationtests/testsuite/administration/AbstractAdministrationTestSuite.java

```diff
@@ -0,0 +1,189 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.integrationtests.testsuite.administration;
+
+import org.axonframework.commandhandling.CommandBus;
+import org.axonframework.commandhandling.CommandHandlingComponent;
+import org.axonframework.commandhandling.gateway.CommandGateway;
+import org.axonframework.configuration.Configuration;
+import org.axonframework.eventsourcing.configuration.EventSourcingConfigurer;
+import org.axonframework.integrationtests.testsuite.administration.commands.AssignTaskCommand;
+import org.axonframework.integrationtests.testsuite.administration.commands.ChangeEmailAddress;
+import org.axonframework.integrationtests.testsuite.administration.commands.CompleteTaskCommand;
+import org.axonframework.integrationtests.testsuite.administration.commands.CreateCustomer;
+import org.axonframework.integrationtests.testsuite.administration.commands.CreateEmployee;
+import org.axonframework.integrationtests.testsuite.administration.commands.GiveRaise;
+import org.axonframework.integrationtests.testsuite.administration.common.PersonIdentifier;
+import org.axonframework.integrationtests.testsuite.administration.common.PersonType;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+import org.junit.jupiter.api.*;
+
+import java.util.concurrent.CompletionException;
+import java.util.function.Function;
+
+/**
+ * Test suite for verifying polymorphic behavior of entities. Can be implemented by different test classes that verify
+ * different ways of building the {@link org.axonframework.modelling.entity.EntityCommandHandlingComponent}.
+ */
+public abstract class AbstractAdministrationTestSuite {
+
+    private static final CreateEmployee CREATE_EMPLOYEE_1_COMMAND = new CreateEmployee(
+            new PersonIdentifier(PersonType.EMPLOYEE, "1234"),
+            "Herrijgers",
+            "Mitchell",
+            "mitchell.herrijgers@axoniq.io",
+            "Bug Creator",
+            3000.0);
+
+
+    private static final CreateCustomer CREATE_CUSTOMER_1_COMMAND = new CreateCustomer(
+            new PersonIdentifier(PersonType.CUSTOMER, "shomer"),
+            "Simpson",
+            "Homer",
+            "homer@the-simpsons.io"
+    );
+
+    private CommandHandlingComponent component;
+    private CommandGateway commandGateway;
+
+    abstract CommandHandlingComponent getCommandHandlingComponent(Configuration configuration);
+
+    @BeforeEach
+    void setUp() {
+        doSetupFor(AbstractAdministrationTestSuite.this::getCommandHandlingComponent);
+    }
+
+    @Test
+    void canNotCreateDuplicateEmployee() {
+        commandGateway.send(CREATE_EMPLOYEE_1_COMMAND, ProcessingContext.NONE).getResultMessage().join();
```

Comment:
Suggestion:
`.getResultMessage().join()` is repeated a lot. You may extract it, but it's just a preference. 

```suggestion
        waitForResult(commandGateway.send(CREATE_EMPLOYEE_1_COMMAND, ProcessingContext.NONE));
        
   private void waitForResult(CommandResult result) {
        result.getResultMessage().join();
    }

```

Reply:
I abstracted it away, nice find